src/Api/Serializer/RestSerializer.php

```diff
@@ -127,6 +127,12 @@ private function applyHeader($name, Shape $member, $value, array &$opts)
             $value = TimestampShape::format($value, 'rfc822');
         }
 
+        if (!empty($member->offsetGet('jsonvalue'))
+            && $member->offsetGet('jsonvalue')
+        ) {
+            $value = base64_encode(json_encode(json_decode($value)));
```

Comment:
I can wrap this around an exception, is that what you meant? This would throw an exception as of now if the json string is not valid.

Also, this is the most concise way to remove all newline characters, tabs etc and make sure that the string we are encoding can be decoded back.